### ISSUE TITLE
Stopped working at Windows CLIs (cmd and powershell)

### ISSUE THREAD

**Issue by nirvor** [KillianLucas/open-interpreter#4]
Hi!

After the (at least?) last update (08.08.23) the open-interpreter stopped working. Basically its not responding to the CLI chat input (or is not starting an openai API request?). This counts for a Windows OS (or most probable Winows File system) and is independent of the CLI (powershell or normal cmd behave identical) but(!) its working as usual at a Ubuntu OS. I attached the CLI behaviour.

How it looks like @ ubuntu:
[https://ibb.co/LPKJYrc](https://ibb.co/LPKJYrc)

How it looks like @ Windows Powershell CLI (normal cmd terminal, too):
[https://ibb.co/s6Lv25g](https://ibb.co/s6Lv25g)

If I had to guess it might again have something to do with the different paths / import os / between Win & Linux?

**Comment by nirvor** [comment]
Small update: I tested it via codespaces, no issues at all, too. I tried to debug it a bit. Was, after checking the interpreter.py, thinking it might be some prompt formatting stuff in cli.py. But seems to be right...

**Comment by nirvor** [comment]
Another update, just trying to help:

Not working @ Win11 via jupyter notebook, same behaviour, see screenshot

Maybe (anyway :-D ) due to a change of the cli.py or any other script regarding the OS CLI interaction function? Would be suprised if the pipeline openinterpreter -> open ai API -> openinterpreter -> chat cli would behave now differently between macos/unix and win OS

![jupyter](https://github.com/KillianLucas/open-interpreter/assets/118377825/91a79e7d-4bbf-4ccb-b1df-c9944c5753a2)

**Comment by nirvor** [comment]
Sooo its working @ google collab boards, too, see snapshot.
![collab](https://github.com/KillianLucas/open-interpreter/assets/118377825/5155a0ca-b3db-4ac5-b47c-8a0debc84820)

Maybe its okay to narrow it down to small recent changes in the cli communication which the windows cli system/syntax/formatting does not really like :-D 

Btw: There might be the option to "skip" that part of debugging if its in actual "CLI functionality" controlable via html/css/streamlit/whatever web interface control, ( so the OS of the users machine should have very little "impact" )

**Comment by KillianLucas** [comment]
Try the latest PIP release— I tested a few things out and put them on PIP since it's so easy to use the package on different computers once I do that. I won't do that anymore because it puts the PIP release out of sync with Github! Now that other folks are working on it, I'll do proper PIP releases, changelogs, etc.

That may or may not have been the problem though. Per the insight of 24fb69a91b8f7a971941b5905bcc776e149a363c I also just changed `os.name` -> `platform.system()` in`code_interpreter.py`, perhaps that will fix this?

**Comment by KillianLucas** [comment]
Also you're right, that would be a great benefit of a web interface (less OS dependent)!

Btw perhaps we should have a `--debug` mode which would print the request to OpenAI and a few other key steps (like the code after its been processed) so we can debug a little easier.

**Comment by KillianLucas** [comment]
Hey @nirvor is this still happening? Hopefully some recent changes have resolved this.

**Comment by nirvor** [comment]
Hey Killian,

just returned from the U.S. back to Ger. I work as a nuclear physicist with
own research group and often things turn... (unnecessary) hectic :_D But
now there should be bit more time.

Iam maybe not super bad in data analysis, but in coding you(!) are for
(dead) sure much more skilled as me..., so most likely I just can help a
bit debugging at this(!) point and can, if you want, can tell you some
"nice to have"  :-D
Anyway: Stick to that(!)  Nothing "bothered" ( :- D) me more in recent
times as the capabilities of these LLMs on economy/science/society since
the "open public" web.

Best wishes,
Tom



Am Di., 5. Sept. 2023 um 01:23 Uhr schrieb killian ***@***.***

**Comment by nirvor** [comment]
Hi Killian,

iam really delighted for you that so many people already appreciate your
work! These are certainly interesting times for you, no doubt :-)

Iam relatively sure that iam slowing down your process if I try to help you
with straight software dev :-D  So, iam not entirely certain how to
contribute to your cool approach without adding any extra stress, apart
from debugging or pointing out specific issues.

-->

Maybe it is quite funny for you from which "corner" i and my interest in
LLMs via CLIs actually comes from... (cough)

Regarding that "nice to have" of the previous msg and to give it some kind
of frame:

In terms of research, my research group and I [2] are primarily active in
experimental (astro)physics R&D, particularly in the development of new
large-scale astrophysical observatories, detectors, and telescopes. Given
that we deploy our telescope arrays in "sky-friendly" environments (high
altitudes, minimal artificial light, and low RFI noise, essentially
"remote" regions, deserts, islands...), the ability to "quickly drive there
and fix stuff" is often limited. Consequently, most of our telescopes,
detectors, observatories, and their DAQ machines are operated, monitored,
and maintained via ssh  tunneling - essentially, via the CLI! 😄

Our main "large scale" observatory we are contributing in R&D, deployment
and data analysis is located at a maybe quite prominent place[1] and we are
currently in the early stages of prototyping demonstration detectors and
telescopes for the next generation of this observatory. After that LLMs and
its future impact "hit" me (I think it was ChatGPT with GPT3) and some
"annoying" API coding work with photosensor interfaces the following day I
started to play- and look around whats already possible in integrating LLMs
into the CLI. And stumbled over your first approaches :-D

At the moment its all still largely in my head (and needs accordingly on my
part still some "PI" work towards colleagues and other research groups) ,
but iam starting to "nudge" the international collaboration of the
observatorium a bit towards all the possible LLM applications in science,
maybe here not through new ways of or more efficient analysis of the
measurement data, which is what primarily suggests itself with LLMs, but a
bit more... more present.

So, the basic idea is to utilize the open-interpreter to operate and
perform tasks on a larger prototype detector - deployed on-site.

I think that would not be the worst possible way of showing off an
"application" and showcasing the open-interpreter. Moreover, maybe in later
years a quite funny story/anecdote for you to tell :-D

Sure, that would probably imply one or the other upgrade on the part of the
functionality of the open-interpreter which might not be on your "roadmap",
such as the possibility of "training" the (in that application of course)
local LL models with e.g. the API reference, technical docu, more detailed
knowledge of microelectronics/DAQ/data processing/etc and a larger
construction site would probably be the security/redundancy of hardware
interfaces and a lot of other stuff but iam already writing quite some time
:-)
In general, i see nothing on the part of the "design decisions" which would
be unpredictably complicated and well, why not "following" now that idea of
first steps in implementing LLMs in astrophysical observatories in parallel
with trying to contribute to your work with your open-interpreter and just
see what develops. I think you have now an idea how i could possibly help /
contribute / tell you what might be needed by guys who are developing cli
controlled DAQs and sensors in general.

Perhaps something in this resonates with you or is worth considering. And,
of course(!), i wont be "disappointed" if not. At the very least, you now
have read about another potential application of LLMs in CLIs and my
underlying motivation for looking into the CLI story in the first place :-)

Best wishes,
Tom


[1] https://icecube.wisc.edu/science/icecube/
[2] Not really updated since at least 5 years + iam not using social media
super much, discord/slack etc is "enough"  :-)
https://www-kseta.ttp.kit.edu/fellows/Thomas.Huber/
https://orcid.org/0000-0002-6515-1673



Am Do., 7. Sept. 2023 um 01:34 Uhr schrieb Thomas Huber <
***@***.***>:

**Comment by jordanbtucker** [comment]
@nirvor Thanks for your interest and willingness to help! Please see the [contributing guidelines](https://github.com/KillianLucas/open-interpreter/blob/main/docs/contributing.md) and come hang out on [the Discord server](https://discord.gg/6p3fD6rBVm)!

**Comment by leifktaylor** [comment]
@jordanbtucker @KillianLucas 
Validated that on WSL2, Windows Command prompt, and MacOS Ventura 13.5.2 that open-interpreter CLI is working as expected.

```
C:\Users\leifkt>pip show open-interpreter
Name: open-interpreter
Version: 0.1.9
```

**I suggest this issue be closed.**
